src/package.ts

```diff
@@ -92,6 +93,39 @@ export class Package extends AsyncOptionalCreatable {
     return versions.includes(this.nextVersion);
   }
 
+  public writePackageJson(rootDir?: string): void {
+    const pkgJsonPath = rootDir ? path.join(rootDir, 'package.json') : 'package.json';
+    fs.writeJsonSync(pkgJsonPath, this.projectJson);
+  }
+
+  public pinDependencyVersions(targetTag: string): ChangedPackageVersions {
+    // get the list of dependencies to hardcode
+    const dependencies: string[] = this.projectJson['pinnedDependencies'];
+    const pinnedPackages = [];
+    dependencies.forEach((name) => {
+      // get the 'release' tag version or the version specified by the passed in tag
+      const result = exec(`npm view ${name} dist-tags --json`, { silent: true });
+      const versions = JSON.parse(result.stdout);
+      let tag = targetTag;
+
+      // if tag is 'latest-rc' and there's no latest-rc release for a package, default to latest
+      if (!versions[tag]) {
+        process.emitWarning(`Warning: no tag ${targetTag} for package: ${name} - default to 'latest'`);
```

Comment:
Now with the output, do we still need this warning? It is clear when there is no tag, because it will show latest in the table.